src/Compiler/Driver/CompilerDiagnostics.fs

```diff
@@ -1795,7 +1794,7 @@ type Exception with
             if s <> "" then
                 os.AppendString(Obsolete2E().Format s)
 
-        | Experimental(s, _) -> os.AppendString(ExperimentalE().Format s)
+        | Experimental(message = s) -> os.AppendString(ExperimentalE().Format s)
```

Comment:
I propose the message includes the added properties if they were present even in plain (text output) build results.

Reply:
Especially the ID might be something useful for searching with.
The url would be better a clicklable, but at least plaintext for command line builds is IMO also beneficial.

Reply:
`Experimental` attribute does not have a message property for  now. It seems that will be added in the next mayor version to match the obsolete attribute See.
https://github.com/dotnet/runtime/pull/108216

But I think we can handle this so it will work when it is available for us.

Reply:
Regarding the `DiagnosticId` We have 2 options:
- Use the provided id from the attribute `[Experimental(diagnosticId: "MY001")]` as a error number which potentially might collide with existing error numbers.
- Use 57 like we do in F# but that will defeat the purpose of the `diagnosticId`

I guess the problem is to report the experimental as error number 57 and then show a diagnostic id was part the error text which will be confusing ?

Reply:
@T-Gro I have updated this PR to follow the same `Obsolete` structures of messages. See tests. Im still not sure about adding the `diagnosticId` as part of the error text as explain in my previous message.